llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -179,6 +179,11 @@ class AMDGPUInformationCache : public InformationCache {
     return {ST.getMinFlatWorkGroupSize(), ST.getMaxFlatWorkGroupSize()};
   }
 
+  SmallVector<unsigned> getMaxNumWorkGroups(const Function &F) {
+    const GCNSubtarget &ST = TM.getSubtarget<GCNSubtarget>(F);
```

Comment:
This function might be generic enough to be put somewhere else if needed.

Reply:
This is attributor boilerplate like the other subtarget queries